async.go

```diff
@@ -0,0 +1,136 @@
+package plc
+
+import (
+	"golang.org/x/sync/errgroup"
+)
+
+type asyncer interface {
+	Wait() error
+	Add(name string, value interface{})
+	AddError(err error)
+}
+
+type action func(string, interface{}) error
+
+type job struct {
+	name  string
+	value interface{}
+}
+
+var noJob = job{}
+
+type async struct {
+	action
+	*errgroup.Group
+	jobs        chan job
+	numRoutines int
+}
+
+func newAsync(act action) async {
+	as := async{
+		action:      act,
+		Group:       &errgroup.Group{},
+		jobs:        make(chan job, 1),
+		numRoutines: 1,
+	}
+
+	as.launchRoutine(noJob)
```

Comment:
I think you could lose the whole newjob semantic, as it's not saving you spawning goroutines.